### ISSUE TITLE
Very large latency in last few weeks

### ISSUE THREAD

**Issue by dlfivefifty** [julia-vscode/julia-vscode#1906]
I've noticed a very significant latency recently, with it taking up to a second to recognise key presses in the terminal. Is this known? 

Making a MWE may be hard but I can try.

**Comment by pfitzseb** [comment]
No, we haven't heard anything about this. Are you on insiders or the normal release?

**Comment by dlfivefifty** [comment]
The normal, latest release. Is there anything I should try first? E.g. delete config files or change to "insiders"?

**Comment by pfitzseb** [comment]
Do you see the latency only for the Julia REPL or for all integrated terminals? If the latter, then the first thing I'd try is the different options for `terminal.integrated.rendererType`. If the former then try disabling the `julia.useRevise` option.

**Comment by dlfivefifty** [comment]
I see it for all of VSCode: even opening files, splitting windows, everything... but only it seems when the Julia REPL is open.

It might be related to another issue I've notice, that printing with very long type signatures is extremely slow: what takes about 1s in standard REPL takes say 30s in VSCode.

**Comment by pfitzseb** [comment]
Hm. Weird. Do you see a Julia process with a heavy CPU load or something like that?

**Comment by marius311** [comment]
Probably not really helpful, but anecdotally I think I see something similar (I posted on the Slack about how long it takes for ↻ to show up sometimes). Let me know if there's anything in particular I can look at when I catch it lagging. CPU load doesn't seem to be it. Maybe it coincides with switching to 1.6?

**Comment by pfitzseb** [comment]
That *should* be a different problem in that it's just the LS crashing and restarting, which can take like 20s or so.

**Comment by marius311** [comment]
Can the LS crash without a log message? I definitely don't see one on the output log. I'll check `top` next time it happens.

**Comment by marius311** [comment]
Ok, I'm almost sure I caught it, and its just that inline evaluation freezes while it says "indexing packages". Maybe I'm crazy, but I don't remember this freezing inline evaluation? I don't exactly remember tbh.

**Comment by dlfivefifty** [comment]
I haven’t noticed the issue since upgrading to Julia v1.6-rc1 so I’m happy to close it

**Comment by jebej** [comment]
I am on v1.5 and I definitely notice that, in a fresh window, despite waiting for "starting language server" message to disappear, and even starting the julia prompt manually with `alt+j+o`, the arrow, and therefore execution of the code, can take 30 seconds.

If I look at the "language server output" tab during that time, I see warning messages about a few "packages not stored on disk" appear slowly one after the other, which I assume means the symbol server is going through all the packages. Once that is complete, the message "Received new data from Julia Symbol Server." appears, and things work from there.

**Comment by dalarev** [comment]
Upgrading to Julia v1.6-rc1 also mitigated a lot of these problems for me, FWIW.

**Comment by marius311** [comment]
Fyi, I opened https://github.com/julia-vscode/julia-vscode/issues/1962 to address one specific aspect thats causing latency.
